scripts/build-types.mjs

```diff
@@ -45,7 +56,30 @@ export async function buildTypes(mapping = defaultMapping) {
              */
             `
         });
-        const content = typescript.replace(/\r\n/g, '\n')
+        let content = typescript.replace(/\r\n/g, '\n')
+
+        // for the typed messages, apply a module augmentation
+        // TODO: add some tests+docs for this. It's a convention based approach to keep JSON schema files compliant
+        if (manifest.kind === 'messages') {
+            const json = JSON.parse(readFileSync(manifest.schema, 'utf8'));
+            const base = basename(manifest.schema, '.json');
+            const relativePath = '../features/' + base + '.js';
+            const className = json.title.replace('Messages', '');
+            const notifications = json.properties?.notifications?.oneOf?.length ?? 0 > 0;
+            const requests = json.properties?.requests?.oneOf?.length ?? 0 > 0;
+            const lines = [];
+            if (notifications) lines.push(`notify: MessagingBase<${json.title}>['notify']`)
+            if (requests) lines.push(`request: MessagingBase<${json.title}>['request']`)
+            const template = `
+declare module ${JSON.stringify(relativePath)} {
+  export interface ${className} {
+    ${lines.join(',   ')}
+  }
+}
+
+`
+            content = content + template;
+        }
```

Comment:
I'll be adding some tests for this code-gen block. It's not very complicated, but since it uses a convention-over-configuration approach (inferring things like class name) I'd like to have a few test cases in place